include/boost/url/grammar/impl/range_rule.hpp

```diff
@@ -372,9 +372,9 @@ class range<T>::
 
     iterator() = default;
     iterator(
-        iterator const&) noexcept = default;
+        iterator const&) = default;
```

Comment:
copies can throw? since when?

Reply:
That's an error we had with GCC 4.8 and Clang 3.8:

```
./boost/url/grammar/impl/range_rule.hpp:374:5: error: function 'boost::urls::grammar::range<T>::iterator::iterator(const boost::urls::grammar::range<T>::iterator&) [with T = boost::urls::experimental::detail::segment_template]' defaulted on its first declaration with an exception-specification that differs from the implicit declaration 'boost::urls::grammar::range<boost::urls::experimental::detail::segment_template>::iterator::iterator(const boost::urls::grammar::range<boost::urls::experimental::detail::segment_template>::iterator&)'
     iterator(
     ^
./boost/url/grammar/impl/range_rule.hpp:376:15: error: function 'boost::urls::grammar::range<T>::iterator& boost::urls::grammar::range<T>::iterator::operator=(const boost::urls::grammar::range<T>::iterator&) [with T = boost::urls::experimental::detail::segment_template]' defaulted on its first declaration with an exception-specification that differs from the implicit declaration 'boost::urls::grammar::range<boost::urls::experimental::detail::segment_template>::iterator& boost::urls::grammar::range<boost::urls::experimental::detail::segment_template>::iterator::operator=(const boost::urls::grammar::range<boost::urls::experimental::detail::segment_template>::iterator&)'
     iterator& operator=(
               ^
```

From this answer

https://stackoverflow.com/questions/47032077/noexcept-default-compilation-error

my understanding is the default declarations can throw if the members can throw and the default declaration will no `noexcept` by default whenever this is possible.

But the compiler doesn't include much detail about the members that are causing the error.

Reply:
we need to figure out why it thinks that the members can throw, because I took time to make sure they didn't. I might have forgotten "noexcept" on one of the dependent types.

Reply:
it has to be `result<T> rv_;`

Reply:
Yes. Definitely.